python/mxnet/gluon/block.py

```diff
@@ -487,8 +487,8 @@ def __init__(self, outputs, inputs, params=None):
         self._params = ParameterDict('', params)
         if isinstance(inputs, symbol.Symbol) and len(inputs.list_outputs()) == 1:
             inputs = [inputs]
-        if isinstance(outputs, symbol.Symbol) and len(outputs.list_outputs()) == 1:
-            outputs = [outputs]
+        if isinstance(outputs, (list, tuple)) and len(outputs) == 1:
+            outputs = outputs[0]
```

Comment:
I actually find the inconsistent type a bit tricky to deal with. For example
```python
In [3]: type(mx.nd.ones((3,5)).split(3, axis=0))
Out[3]: list

In [4]: type(mx.nd.ones((3,5)).split(1, axis=0))
Out[4]: mxnet.ndarray.ndarray.NDArray
```

Pytorch does it more consistently. `torch.chunk` always returns a tuple even when length is 1.

Reply:
That's a different matter. This is block

Reply:
Isn't this logic unwrapping a list of outputs?

Reply:
If block returns a list you need to do `y = layer(x)[0]`